### DISCUSSION TITLE
Custom citation key format

### DISCUSSION THREAD

**Discussion by alexreg** [retorquere/zotero-better-bibtex#1932] [Q&A] [answered]
I would a citation key format like the following:

* `Smith:1990`
* `Smith;Brown:2000`
* `Smith;ea:2010`

Is this possible?

Note, surnames should retain capitalisation, though non-ASCII characters should be converted to ASCII equivalents. It would also be interesting to have a variant where prefixes like "de", "von", "van", etc. are prepended to the surname, e.g., "vonNeumann" (von Neumann, John), "tHooft" ('t Hooft, Gerardus) — but that's just a bonus.

**Comment by retorquere** [comment, marked as answer]
`[authors2+;]:[year]`

**Reply by retorquere** [reply]
I'd have to think about prefixes -- that's currently not possible. ASCII-fication is the default behavior.

**Reply by alexreg** [reply]
Thanks, that's almost it! Two small issues:
a) Can I get it to use "ea" instead of "EtAl"?
b) It displays two authors *and* "EtAl" when I have three authors, whereas I want to always cap it at two "authors". i.e., "AuthorA;AuthorB" for two, "AuthorA;ea" for three or more.

**Reply by retorquere** [reply]
a) `[authors2+;:replace=etal,ea]:[year]`
b) is a reasonable addition, but you'd have to open an issue for that, and it will be at least a couple of days, maybe a week, before I can put any work in it, as I'm currently working on 3 issues in parallel already.

**Reply by alexreg** [reply]
No worries. Thanks for the quick replies.

**Reply by alexreg** [reply]
Oh, incidentally, that won't cause any problems if an author name just so happens to contain "etal" somewhere within, will it? (I don't know exactly how the `replace` thing works.)

**Reply by retorquere** [reply]
That would be a problem. `:replace=;etal$,ea,regex` should work except when the author is actually named `etal`. If it's really to replace the `etal` as a unit, I'd have to extend the `authors` syntax. Not off the table necessarily.
